src/models.jl

```diff
@@ -0,0 +1,156 @@
+"""
+    Models for random hypergraphs building:
+        * random model
+        * k-uniform model
+        * d-uniform model
+        * preferential-attachment
+"""
+
+
+"""
+    random_model(nVertices, nEdges)
+
+Generate a *random* hypergraph without any structural constraints.
+
+**The Algorithm**
+Given two integer parameters *nVertices* and *nEdges* (the number of nodes and hyperedges, respectively),
+the algorithm computes - for each hyperedge *he={1,...,m}* -
+a random number *s ϵ [1, n]* (i.e. the hyperedge size).
+Then, the algorithm selects uniformly at random *s* vertices from *V* to be added in *he*.
+"""
+function random_model(nVertices::Int, nEdges::Int)
+    mx = Matrix{Union{Nothing,Bool}}(nothing, nVertices,nEdges)
+    for e in 1:size(mx,2)
+        nv = rand(1:size(mx,1))
+        mx[sample(1:size(mx,1), nv;replace=false), e] .= true
+    end
+
+    h = Hypergraph(mx)
+    if all(length.(h.v2he) .> 0)
+        return h
+    else
+        return random_model(nVertices, nEdges)
+    end
+end
+
+
+"""
+    random_kuniform_model(nVertices, nEdges, k)
+
+Generates a *k*-uniform hypergraph, i.e. an hypergraph where each hyperedge has size *k*.
+
+**The Algorithm**
+The algorithm proceeds as the *randomH*, forcing the size of each hyperedge equal to *k*.
+"""
+function random_kuniform_model(nVertices::Int, nEdges::Int, k::Int)
+    mx = Matrix{Union{Nothing,Bool}}(nothing, nVertices,nEdges)
+    for e in 1:size(mx,2)
+        nv = k
+        mx[sample(1:size(mx,1), nv;replace=false), e] .= true
+    end
+
+    h = Hypergraph(mx)
+    if all(length.(h.he2v) .== k)
+        return h
+    else
+        return random_kuniform_model(nVertices, nEdges, k)
+    end
+end
+
+
+"""
+    random_dregular_model(nVertices, nEdges, d)
+
+Generates a *d*-regular hypergraph, where each node has degree *d*.
+
+**The Algorithm**
+The algorithm exploits the *k*-uniform approach described for the *random_kuniform_model* method
+to build a *d*-regular hypergraph *H* having *nVertices* nodes and *nEdges* edges.
+It returns the hypergraph H^* dual of *H*.
+"""
+function random_dregular_model(nVertices::Int, nEdges::Int, d::Int)
+    mx = Matrix{Union{Nothing,Bool}}(nothing, nVertices,nEdges)
+
+    for v in 1:size(mx,1)
+        ne = d
+        mx[v, sample(1:size(mx,2), ne;replace=false)] .= true
+    end
+
+    h = Hypergraph(mx)
+    if all(length.(h.v2he) .== d)
```

Comment:
again - this probably should be an assertion.

Reply:
The same for the random model, but here we are looking for a random graph with each node has degree equal to d

Reply:
again - is this possible that this check fails?

Reply:
same